pkg/controller/resource/deploymentconfig.go

```diff
@@ -0,0 +1,70 @@
+package resource
+
+import (
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	chaosTypes "github.com/litmuschaos/chaos-operator/pkg/controller/types"
+)
+
+var (
+	gvr = schema.GroupVersionResource{
+		Group:    "apps.openshift.io",
+		Version:  "v1",
+		Resource: "deploymentconfigs",
+	}
+)
+// CheckDeploymentAnnotation will check the annotation of deployment
+func CheckDeploymentConfigAnnotation(clientSet dynamic.Interface, engine *chaosTypes.EngineInfo) (*chaosTypes.EngineInfo, error) {
+
+	deploymentConfigList, err := getDeploymentConfigList(clientSet, engine)
+	if err != nil {
+		return engine, err
+	}
+	engine, chaosEnabledDeploymentConfig, err := checkForChaosEnabledDeploymentConfig(deploymentConfigList, engine)
+	if err != nil {
+		return engine, err
+	}
+
+	if chaosEnabledDeploymentConfig == 0 {
+		return engine, errors.New("no DeploymentConfig chaos-candidate found")
+	}
+	chaosTypes.Log.Info("DeploymentConfig chaos candidate:", "appName: ", engine.AppName, " appUUID: ", engine.AppUUID)
+
+	return engine, nil
+}
+
+func getDeploymentConfigList(clientSet dynamic.Interface, engine *chaosTypes.EngineInfo) (*unstructured.UnstructuredList, error) {
+
+	dynamicClient := clientSet.Resource(gvr)
+
+	deploymentConfigList, err := dynamicClient.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error while listing deploymentconfigs with matching labels %s", engine.Instance.Spec.Appinfo.Applabel)
+	}
+	if len(deploymentConfigList.Items) == 0 {
+		return nil, fmt.Errorf("no deploymentconfigs with matching labels %s", engine.Instance.Spec.Appinfo.Applabel)
+	}
+	return deploymentConfigList, err
+}
+
+// This will check and count the total chaos enabled application
+func checkForChaosEnabledDeploymentConfig(deploymentConfigList *unstructured.UnstructuredList, engine *chaosTypes.EngineInfo) (*chaosTypes.EngineInfo, int, error) {
+
```

Comment:
Can we validate that the GVR of `*unstructured.UnstructuredList` matches with `deploymentConfig` GVR, and then start to validate.

Reply:
Or add the validation in its above function